### ISSUE TITLE
Convert TypeInference lens into an Adaptive Schema

### ISSUE THREAD

**Issue by okennedy** [UBOdin/mimir#193]
When Arindam first created the TypeInference lens several years back, we didn't have Adaptive Schemas.  We do now.  TypeInference is really something that belongs as an AdaptiveSchema.  One nifty consequence is that we might be able to do knowledge inference entirely in the backend db, something along the lines of:
```
CREATE TABLE TYPES(name varchar, regexp varchar);
CREATE LENS TYPE_CHOICE
AS SELECT 'col' as col, types.name, COUNT(*) AS count
   FROM (SELECT col FROM input LIMIT 10000) input, types 
   WHERE regexp_match(types.regexp, input.col))
WITH KEY_REPAIR(col, SCORE_BY(count))
```

**Comment by okennedy** [comment]
Another advantage to doing this is that it should be easier to set things up so that only values that are NULLified are highlighted in red.

**Comment by mrb24** [comment]
Closed by commit: f4364ba66fcd007e9412fc54de8beb50cada891e
